test/integration.basic.js

```diff
@@ -439,6 +439,8 @@ tap.test('integration basic - multiple hosts - mainfest is on one host but conte
 });
 
 tap.test('integration basic - multiple protocols - mainfest is on a http host but content on fallbacks on https hosts', async t => {
+    // Undici rejects self signed SSL certs so we need to disable that for tests
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
```

Comment:
Undici won't accept self signed SSL certs. This line disables that for test purposes. Does this seem acceptable?

Reply:
Did you look at this one: https://github.com/nodejs/undici/blob/main/docs/best-practices/client-certificate.md

It would be nice to have support for self signed SSL certs in Podium as a client.

Reply:
Do you want to try to get that in as part of this or could it be a feature added after?

Reply:
The client does atm support self signed SSL certs by setting `rejectUnauthorized` to `false`: https://github.com/podium-lib/client/blob/master/lib/client.js#L89 so we're not adding a new feature. Though, to me it looks like `request` is a bit more laxed on allowing self signed SSL certs than what `undici` is. In `request` it seems to be enough to just set `rejectUnauthorized` to `false` but in `undici` one need to also pass in the same certificates as the server is using (ref the best practices link posted above).

Preferelly I would like us to continue supporting self signed SSL certs but it would require a slight change in the property so it can also import the certs.

Reply:
The latest commit addresses this. I replaced the request method with a client instance that can take TLS options and was able to support the rejectUnauthorised flag in this way. Likely non optimal to reinstantiate the client on every request but since the client is bound to an origin, we would need to find a way to implement a new client for each origin in play if we want to keep the client around. We could probably do this with a map keyed by origin? Not sure if its worth it at this stage? Something to revisit in a second perf iteration with benchmarks?

Reply:
I think we don't really need to keep track of this our self. Undici has the concept of a balanced pool which can hold client instances connected to different upstreams: https://github.com/nodejs/undici/blob/main/docs/api/BalancedPool.md

Though; I think we should merge this at this point and look at optimizing this a little bit further down the road.